[BUG] (dev/3.0.0) Invalid AI configuration breaks suggestions endpoint

### Description

When the AI configuration is invalid, for example if an unsupported model is selected, the suggestions endpoint (`/api/documents/:id/suggestions`) fails with an internal server error and status code 500.

### Steps to reproduce

1. Configure the AI integration with invalid parameters (e.g. unsupported model name)
2. Try to get suggestions for a document

### Webserver logs

```bash
webserver-1  | [2026-04-25 14:01:49,001] [ERROR] [django.request] Internal Server Error: /api/documents/1/suggestions/
webserver-1  | Traceback (most recent call last):
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 555, in thread_handler
webserver-1  |     raise exc_info[1]
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/django/core/handlers/exception.py", line 42, in inner
webserver-1  |     response = await get_response(request)
webserver-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 555, in thread_handler
webserver-1  |     raise exc_info[1]
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/django/core/handlers/base.py", line 253, in _get_response_async
webserver-1  |     response = await wrapped_callback(
webserver-1  |                ^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 504, in __call__
webserver-1  |     ret = await asyncio.shield(exec_coro)
webserver-1  |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/current_thread_executor.py", line 40, in run
webserver-1  |     result = self.fn(*self.args, **self.kwargs)
webserver-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/asgiref/sync.py", line 559, in thread_handler
webserver-1  |     return func(*args, **kwargs)
webserver-1  |            ^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
webserver-1  |     return view_func(request, *args, **kwargs)
webserver-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/viewsets.py", line 125, in view
webserver-1  |     return self.dispatch(request, *args, **kwargs)
webserver-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 515, in dispatch
webserver-1  |     response = self.handle_exception(exc)
webserver-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 475, in handle_exception
webserver-1  |     self.raise_uncaught_exception(exc)
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 486, in raise_uncaught_exception
webserver-1  |     raise exc
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/rest_framework/views.py", line 512, in dispatch
webserver-1  |     response = handler(request, *args, **kwargs)
webserver-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/django/utils/decorators.py", line 48, in _wrapper
webserver-1  |     return bound_method(*args, **kwargs)
webserver-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/django/views/decorators/cache.py", line 54, in _view_wrapper
webserver-1  |     response = viewfunc(request, *args, **kw)
webserver-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/django/utils/decorators.py", line 48, in _wrapper
webserver-1  |     return bound_method(*args, **kwargs)
webserver-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/django/views/decorators/http.py", line 154, in inner
webserver-1  |     response = func(request, *args, **kwargs)
webserver-1  |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/src/paperless/src/documents/views.py", line 1327, in suggestions
webserver-1  |     llm_suggestions = get_ai_document_classification(doc, request.user)
webserver-1  |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/src/paperless/src/paperless_ai/ai_classifier.py", line 101, in get_ai_document_classification
webserver-1  |     result = client.run_llm_query(prompt)
webserver-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/src/paperless/src/paperless_ai/client.py", line 68, in run_llm_query
webserver-1  |     result = self.llm.chat_with_tools(
webserver-1  |              ^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/llama_index/core/llms/function_calling.py", line 46, in chat_with_tools
webserver-1  |     chat_kwargs = self._prepare_chat_with_tools_compat(
webserver-1  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/llama_index/core/llms/function_calling.py", line 158, in _prepare_chat_with_tools_compat
webserver-1  |     return self._prepare_chat_with_tools(
webserver-1  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/llama_index_instrumentation/dispatcher.py", line 335, in wrapper
webserver-1  |     result = func(*args, **kwargs)
webserver-1  |              ^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/llama_index/llms/openai/base.py", line 999, in _prepare_chat_with_tools
webserver-1  |     if self.metadata.is_function_calling_model:
webserver-1  |        ^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/llama_index/llms/openai/base.py", line 385, in metadata
webserver-1  |     context_window=openai_modelname_to_contextsize(self._get_model_name()),
webserver-1  |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
webserver-1  |   File "/usr/local/lib/python3.12/site-packages/llama_index/llms/openai/utils.py", line 383, in openai_modelname_to_contextsize
webserver-1  |     raise ValueError(
webserver-1  | ValueError: Unknown model 'gpt-5.4-mini-2026-03-17'. Please provide a valid OpenAI model name in: o1, o1-2024-12-17, o1-pro, o1-pro-2025-03-19, o1-preview, o1-preview-2024-09-12, o1-mini, o1-mini-2024-09-12, o3-mini, o3-mini-2025-01-31, o3, o3-2025-04-16, o3-pro, o3-pro-2025-06-10, o4-mini, o4-mini-2025-04-16, gpt-5, gpt-5-2025-08-07, gpt-5-chat, gpt-5-chat-latest, gpt-5-mini, gpt-5-mini-2025-08-07, gpt-5-nano, gpt-5-nano-2025-08-07, gpt-5-pro, gpt-5-pro-2025-10-06, gpt-5.1, gpt-5.1-2025-11-13, gpt-5.1-chat-latest, gpt-5.2, gpt-5.2-2025-12-11, gpt-5.2-chat-latest, gpt-5.3, gpt-5.3-chat-latest, gpt-5.4, gpt-5.4-2026-03-05, gpt-5.4-mini, gpt-5.4-nano, gpt-5.4-chat-latest, gpt-4, gpt-4-32k, gpt-4-1106-preview, gpt-4-0125-preview, gpt-4-turbo-preview, gpt-4-vision-preview, gpt-4-1106-vision-preview, gpt-4-turbo-2024-04-09, gpt-4-turbo, gpt-4o, gpt-4o-audio-preview, gpt-4o-audio-preview-2024-12-17, gpt-4o-audio-preview-2024-10-01, gpt-4o-mini-audio-preview, gpt-4o-mini-audio-preview-2024-12-17, gpt-4o-2024-05-13, gpt-4o-2024-08-06, gpt-4o-2024-11-20, gpt-4.5-preview, gpt-4.5-preview-2025-02-27, chatgpt-4o-latest, gpt-4o-mini, gpt-4o-mini-2024-07-18, gpt-4-0613, gpt-4-32k-0613, gpt-4-0314, gpt-4-32k-0314, gpt-4.1, gpt-4.1-mini, gpt-4.1-nano, gpt-4.1-2025-04-14, gpt-4.1-mini-2025-04-14, gpt-4.1-nano-2025-04-14, gpt-3.5-turbo, gpt-3.5-turbo-16k, gpt-3.5-turbo-0125, gpt-3.5-turbo-1106, gpt-3.5-turbo-0613, gpt-3.5-turbo-16k-0613, gpt-3.5-turbo-0301, text-davinci-003, text-davinci-002, gpt-3.5-turbo-instruct, text-ada-001, text-babbage-001, text-curie-001, ada, babbage, curie, davinci, gpt-35-turbo-16k, gpt-35-turbo, gpt-35-turbo-0125, gpt-35-turbo-1106, gpt-35-turbo-0613, gpt-35-turbo-16k-0613
```

### Browser logs

```bash

```

### Services logs

```bash

```

### Paperless-ngx version

latest dev

### Host OS

macOS

### Installation method

Docker - official image

### System status

```json

```

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/src/documents/tests/test_views.py
+++ b/src/documents/tests/test_views.py
@@ -356,6 +356,31 @@ class TestAISuggestions(DirectoriesMixin, TestCase):
             },
         )
 
+    @patch("documents.views.get_ai_document_classification")
+    @override_settings(
+        AI_ENABLED=True,
+        LLM_BACKEND="openai",
+    )
+    def test_suggestions_with_invalid_ai_configuration(
+        self,
+        mock_get_ai_classification,
+    ) -> None:
+        mock_get_ai_classification.side_effect = ValueError(
+            "Unknown model 'gpt-5.4-mini-2026-03-17'.",
+        )
+
+        self.client.force_login(user=self.user)
+        response = self.client.get(f"/api/documents/{self.document.pk}/suggestions/")
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(
+            response.json(),
+            {
+                "ai": ["Invalid AI configuration."],
+            },
+        )
+        self.assertIsNone(get_llm_suggestion_cache(self.document.pk, backend="openai"))
+
     def test_invalidate_suggestions_cache(self) -> None:
         self.client.force_login(user=self.user)
         suggestions = {

--- a/src/documents/views.py
+++ b/src/documents/views.py
@@ -1320,7 +1320,17 @@ class DocumentViewSet(
                 refresh_suggestions_cache(doc.pk)
                 return Response(cached_llm_suggestions.suggestions)
 
-            llm_suggestions = get_ai_document_classification(doc, request.user)
+            try:
+                llm_suggestions = get_ai_document_classification(doc, request.user)
+            except ValueError as exc:
+                logger.exception(
+                    "Invalid AI configuration while generating suggestions for "
+                    "document %s: %s",
+                    doc.pk,
+                    exc,
+                    exc_info=True,
+                )
+                raise ValidationError({"ai": [_("Invalid AI configuration.")]}) from exc
 
             matched_tags = match_tags_by_name(
                 llm_suggestions.get("tags", []),

